spec/ndx-patterned-ogen.extensions.yaml

```
@@ -169,46 +251,55 @@ groups:
       - name: targeted_rois
         neurodata_type_inc: DynamicTableRegion
         doc: A table region referencing a PlaneSegmentation object storing targeted ROIs.
-        required: true
       - name: segmented_rois
         neurodata_type_inc: DynamicTableRegion
         doc: A table region referencing a PlaneSegmentation object storing segmented ROIs that receive photostimulation.
-        required: false
+        quantity: "?"
   - neurodata_type_def: PatternedOptogeneticStimulusTable
     neurodata_type_inc: TimeIntervals
     doc: Table to hold all patterned optogenetic stimulus onsets.
-    quantity: "?"
     datasets:
       - name: power
         neurodata_type_inc: VectorData
-        doc: Power (in Watts) applied to each target during patterned photostimulation.
-        required: true
+        doc: Power applied to each target during patterned photostimulation, in W.
+        attributes:
+        - name: unit
+          value: Watts
+          doc: unit of measure of power which is fixed to Watts
+          dtype: text
       - name: targets
         neurodata_type_inc: VectorData
         dtype:
           target_type: OptogeneticStimulusTarget
           reftype: object
         doc: Targeted rois for the stimulus onset.
-        required: true
       - name: stimulus_pattern
         neurodata_type_inc: VectorData
         dtype:
           target_type: LabMetaData
           reftype: object
         doc: Link to the stimulus pattern.
-        required: true
       - name: stimulus_site
         neurodata_type_inc: VectorData
         dtype:
           target_type: PatternedOptogeneticStimulusSite
           reftype: object
         doc: Link to the stimulus site.
-        required: true
       - name: frequency
         neurodata_type_inc: VectorData
-        doc: Frequency of stimulation if the stimulus delivered is pulsed (in Hz).
-        required: false
+        doc: Frequency of stimulation if the stimulus delivered is pulsed, in Hz.
+        quantity: "?"
+        attributes:
+        - name: unit
+          value: Hertz
+          doc: unit of measure of frequency which is fixed to Hertz
+          dtype: text
       - name: pulse_width
         neurodata_type_inc: VectorData
         doc: Pulse width of stimulation if the stimulus delivered is pulsed, in seconds/phase.
-        required: false
+        quantity: "?"
+        attributes:
+        - name: unit
+          value: seconds/phase
+          doc: unit of measure of power which is fixed to seconds/phase
```

```suggestion
          doc: Unit of measure of power, fixed to seconds per phase.
```

I like the approach where the doc expands the symbolic representations of the complex SI form in pure english